gpflow/kernels/base.py

```diff
@@ -36,25 +36,28 @@ def __init__(self,
                  active_dims: Optional[Union[slice, list]] = None,
                  name: Optional[str] = None):
         """
-        :param active_dims: active dimensions, has the slice type.
+        :param active_dims: active dimensions, either a slice or list of
+            indices into the columns of X.
         :param name: optional kernel name.
         """
         super().__init__(name=name)
-        if isinstance(active_dims, list):
-            active_dims = np.array(active_dims)
-        self._active_dims = active_dims
+        self._active_dims = self._normalize_active_dims(active_dims)
+
+    @staticmethod
+    def _normalize_active_dims(value):
```

Comment:
From the name is not obvious what the function does.

Reply:
Suggestions for a better name welcome. Is "canonize" better? How do you call the thing that takes a bunch of inputs and makes sure regardless of what it started out with it's now a more limited form so that we don't have to special-case everywhere?

Reply:
Let me think...